pkg/operator/ceph/file/dependent.go

```diff
@@ -55,8 +58,18 @@ func cephFilesystemDependents(clusterdCtx *clusterd.Context, clusterInfo *client
 	nsName := fmt.Sprintf("%s/%s", filesystem.Namespace, filesystem.Name)
 	baseErrMsg := fmt.Sprintf("failed to get dependents of CephFilesystem %q", nsName)
 
+	deps := dependents.NewDependentList()
+	_, err := cephclient.GetFilesystem(clusterdCtx, clusterInfo, filesystem.Name)
+	if err != nil {
+		if code, ok := kexec.ExitStatus(err); ok && code == int(syscall.ENOENT) {
+			logger.Infof("the deletion will continue without checking for dependencies since the filesystem CR was not found %q", nsName)
```

Comment:
This comment is not correct. This does not check if the CephFilesystem CR exists, it checks if Ceph has a record of the filesystem.

```suggestion
			logger.Infof("filesystem %q deletion will continue without checking for dependencies since the the filesystem does not exist within Ceph", nsName)
```